### PR TITLE
ocs-to-ocs: Add scripts to install dependencies and auto generate protobuf files

### PR DESCRIPTION
**Pull request by sp98**
This PR:
-  Updated makefile and added new script /hack/gen-protobuf.sh.
This script will install protoc, protoc-gen-go and protoc-gen-go-grpc
tools in build/_output directory (not checked in) and generate protobuf files. Only the linux
version of the dependencies are downloaded as of now.
- Moved .proto files to separate directory.
- Added the updated protobuf files after running the auto gen scripts

### CODE REVIEW

File: hack/gen-protobuf.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+if [[ $EXISTING_PROTOC_GEN_GO_GRPC != "protoc-gen-go-grpc ${PROTOC_GEN_GO_GRPC_VERSION}" ]]; then
+   # download protoc-gen-go-grpc
+	wget -P ${OUTDIR_PROTO_DIST} --backups=1 \
+		https://github.com/grpc/grpc-go/releases/download/cmd%2Fprotoc-gen-go-grpc%2Fv${PROTOC_GEN_GO_GRPC_VERSION}/protoc-gen-go-grpc.v${PROTOC_GEN_GO_GRPC_VERSION}.linux.386.tar.gz
+	tar -C ${OUTDIR_GRPC} -zxvf ${OUTDIR_PROTO_DIST}/protoc-gen-go-grpc.v${PROTOC_GEN_GO_GRPC_VERSION}.linux.386.tar.gz ./protoc-gen-go-grpc
+fi
+
+
+# services
+declare -a services=("provider")
```
Comment (jarrpa) [inline hack/gen-protobuf.sh:42]: Could you define this in `common.sh`? It feels like it'd be useful for future development.

**Review by jarrpa — CHANGES_REQUESTED** [review]
Overall looks solid.

File: hack/gen-protobuf.sh (hunk 1 above)
Comment (sp98) [reply hack/gen-protobuf.sh]: moved it to `common.sh`

**Review by jarrpa — APPROVED** [review]
/lgtm

**Pull request merged**